packages/versions/src/lib/semver.ts

```diff
@@ -0,0 +1,51 @@
+export class Semver {
+  private major: number;
+  private minor: number;
+  private patch: number;
+
+  private constructor(version: string) {
+    const [major, minor, patch] = version.split('.').map((v) => parseInt(v, 10));
+    this.major = major;
+    this.minor = minor;
+    this.patch = patch;
+  }
+
+  private static compareVersions(version1: string, version2: string): number {
+    const semver1 = new Semver(version1);
+    const semver2 = new Semver(version2);
+
+    if (semver1.major !== semver2.major) {
+      return semver1.major - semver2.major;
+    }
+    if (semver1.minor !== semver2.minor) {
+      return semver1.minor - semver2.minor;
+    }
+    return semver1.patch - semver2.patch;
+  }
+
+  public static gt(version1: string, version2: string): boolean {
+    return this.compareVersions(version1, version2) > 0;
+  }
+
+  public static eq(version1: string, version2: string): boolean {
+    return this.compareVersions(version1, version2) === 0;
+  }
+
+  public static majorEq(version1: string, version2: string): boolean {
+    const semver1 = new Semver(version1);
+    const semver2 = new Semver(version2);
+    return semver1.major === semver2.major;
+  }
+
+  public static minorEq(version1: string, version2: string): boolean {
+    const semver1 = new Semver(version1);
+    const semver2 = new Semver(version2);
+    return semver1.minor === semver2.minor;
+  }
+
+  public static patchEq(version1: string, version2: string): boolean {
+    const semver1 = new Semver(version1);
+    const semver2 = new Semver(version2);
+    return semver1.patch === semver2.patch;
+  }
+}
```

Comment:
Looking good! Two only considerations so far:
 1. The use of classes seems unnecessary — can we use simple functions?
 1. Consider adding a `semver.test.ts` covering 💯 everything

> _Please apply the same mentality to all other similar PRs around tree-shakeability._

```ts
// ALERT: pseudo-code
// Just playing around with your stuff :)

export const parseVersionSegments = (version: string) => {
  const [major, minor, patch] = version.split('.').map((v) => parseInt(v, 10));
  return { major, minor, patch };
};

export type VersionSegment = 'major' | 'minor' | 'patch';

export const isSegmentEq = (v1: string, v2: string, segment: VersionSegment) => {
  const seg1 = parseVersionSegments(v1)[segment];
  const seg2 = parseVersionSegments(v2)[segment];
  return seg1 === seg2;
};

export const diffVersions = (v1: string, v2: string) => {
  const segsA = parseVersionSegments(v1);
  const segsB = parseVersionSegments(v2);
  if (segsA.major !== segsB.major) {
    return segsA.major - segsB.major;
  }
  if (segsA.minor !== segsB.minor) {
    return segsA.minor - segsB.minor;
  }
  return segsA.patch - segsB.patch;
};

export const isEq = (v1: string, v2: string) => diffVersions(v1, v2) === 0;
export const isGt = (v1: string, v2: string) => diffVersions(v1, v2) > 0;

export const isMajorEq = (v1: string, v2: string) => isSegmentEq(v1, v2, 'major');
export const isMinorEq = (v1: string, v2: string) => isSegmentEq(v1, v2, 'minor');
export const isPatchEq = (v1: string, v2: string) => isSegmentEq(v1, v2, 'patch');

```

Reply:
https://github.com/FuelLabs/fuels-ts/pull/1594/commits/29b918957c19b37ce026e1dbfca69b9515a50441

Reply:
https://github.com/FuelLabs/fuels-ts/pull/1594/commits/4a9f53b519087674ef92b32f3f84e6bb917d10ea